### ISSUE TITLE
Define plugin default value generator reuses cached values

### ISSUE THREAD

**Issue by akagomez** [canjs/canjs#2412]
If a default value generator (a `value` method) depends on a property with a defined `Value` property, the created `Value` instance is stored on the constructor's `defaults` property and improperly reused for subsequent instances of the map. 

Here's an example: 

```
var Map = can.Map.extend({
  define: {
    foo: {
      Value: can.Map
    },
    bar: {
      value: function () {
        this.attr('foo');
        return true;
      } 
    }  
  }
});

var map1 = new Map();
var map2 = new Map({
  foo: {}
});

console.log(Map.defaults.foo._cid); //-> ".map4"
console.log(map1.attr('foo')._cid); //-> ".map4"
console.log(map2.attr('foo')._cid); //-> ".map4"
```

And demo: http://jsbin.com/cepodi/edit?html,js,console

**Comment by akagomez** [comment]
I've done some work on default value generators, but I can't remember why we store the resulting default values on the constructor: https://github.com/canjs/canjs/blob/a3b566ceb93dc60e4f1e909f9a27123b4ff55771/map/map.js#L192

**Comment by justinbmeyer** [comment]
@akagomez can you move this to https://github.com/canjs/can-map-define and close it here?

**Comment by justinbmeyer** [comment]
(I accidentally closed it)

**Comment by akagomez** [comment]
Closing because this issue won't be addressed in 2.3. Look for the fix in ^3.0.
